invenio_rdm_records/tokens/resource_access.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Resource access tokens API."""
+
+from datetime import datetime, timedelta
+
+import jwt
+from flask import current_app
+from invenio_oauth2server.models import Token
+
+from .errors import (
+    ExpiredTokenError,
+    InvalidTokenError,
+    InvalidTokenIDError,
+    MissingTokenIDError,
+)
+from .scopes import tokens_generate_scope
+
+
+class ResourceAccessToken:
+    """Resource access tokens."""
+
+    def __init__(self, token):
+        """Initialize RAT token."""
+        self.token = token
+
+    def validate_rat(self):
```

Comment:
I agree, this can be a simple utility as it was in Zenodo. It became a class as i tried to map it to resource/service logic, but it didn't make sense here.